crates/bevy_pbr/src/render/mesh_bindings.rs

```diff
@@ -1,236 +1,348 @@
-        entry(binding, MORPH_BUFFER_SIZE as u64, buffer)
+}
+
+// --- Individual bind group entries ---
+
+fn entry(binding: u32, size: u64, buffer: &Buffer) -> BindGroupEntry {
+    BindGroupEntry {
+        binding,
+        resource: BindingResource::Buffer(BufferBinding {
+            buffer,
+            offset: 0,
+            size: Some(BufferSize::new(size).unwrap()),
+        }),
     }
-    pub(super) fn targets(binding: u32, texture: &TextureView) -> BindGroupEntry {
-        BindGroupEntry {
-            binding,
-            resource: BindingResource::TextureView(texture),
-        }
+}
+fn model_entry(binding: u32, resource: BindingResource) -> BindGroupEntry {
+    BindGroupEntry { binding, resource }
+}
+fn skinning_entry(binding: u32, buffer: &Buffer) -> BindGroupEntry {
+    entry(binding, JOINT_BUFFER_SIZE as u64, buffer)
+}
+fn weights_entry(binding: u32, buffer: &Buffer) -> BindGroupEntry {
+    entry(binding, MORPH_BUFFER_SIZE as u64, buffer)
+}
+fn targets_entry(binding: u32, texture: &TextureView) -> BindGroupEntry {
+    BindGroupEntry {
+        binding,
+        resource: BindingResource::TextureView(texture),
     }
 }
 
+// --- Layout entry lists & bind group entry lists ---
+
+const MVECTORS_OFFSET: u32 = 3;
+
+fn model_layout_entries(device: &RenderDevice) -> [BindGroupLayoutEntry; 1] {
+    [model_layout(device, 0)]
+}
+fn model_entries(model: BindingResource) -> [BindGroupEntry; 1] {
+    [model_entry(0, model)]
+}
+
+fn morph_layout_entries() -> [BindGroupLayoutEntry; 2] {
+    [weights_layout(2), targets_layout(3)]
+}
+fn morph_entries<'a>(weights: &'a Buffer, targets: &'a TextureView) -> [BindGroupEntry<'a>; 2] {
+    [weights_entry(2, weights), targets_entry(3, targets)]
+}
+
+fn skin_layout_entries() -> [BindGroupLayoutEntry; 1] {
+    [skinning_layout(1)]
+}
+fn skin_entries(skin: &Buffer) -> [BindGroupEntry; 1] {
+    [skinning_entry(1, skin)]
+}
+
+fn mvectors_morph_layout_entries() -> [BindGroupLayoutEntry; 1] {
+    [weights_layout(2 + MVECTORS_OFFSET)]
+}
+fn mvectors_morph_entries(weights: &Buffer) -> [BindGroupEntry; 1] {
+    [weights_entry(2 + MVECTORS_OFFSET, weights)]
+}
+
+fn mvectors_skin_layout_entries() -> [BindGroupLayoutEntry; 1] {
+    [skinning_layout(1 + MVECTORS_OFFSET)]
+}
+fn mvectors_skin_entries(skin: &Buffer) -> [BindGroupEntry; 1] {
+    [skinning_entry(1 + MVECTORS_OFFSET, skin)]
+}
+
 /// All possible [`BindGroupLayout`]s in bevy's default mesh shader (`mesh.wgsl`).
 #[derive(Clone)]
-pub struct MeshLayouts {
-    /// The mesh model uniform (transform) and nothing else.
-    pub model_only: BindGroupLayout,
-
-    /// Also includes the uniform for skinning
-    pub skinned: BindGroupLayout,
+pub struct MeshBindings<T> {
+    pub model_only: T,
+    pub morphed: T,
+    pub skinned: T,
+    pub morphed_skinned: T,
+    pub morphed_mvectors: T,
+    pub skinned_mvectors: T,
+    pub morphed_skinned_mvectors: T,
```

Comment:
mvectors -> motion_vectors, and same for other places like the functions above

Reply:
I considered using `motion_vectors` when writing the code, but preferred `mvectors` because it's (1) shorter (2) is visually comparable to `skinned` and `morphed` so that it's easy at a glance to see what are the binding groups available on the given field.

Would adding a comment clarifying that `mvectors` stands for "motion vectors" be enough?

Reply:
I'd personally rather motion_vectors fully written out. Reading code matters much more than writing it. Up to maintainer decision though.

Reply:
longer ≠ more readable. More words to parse is more energy spent to understand code. Furthermore, when the modifiers are `skinned` and `morphed`, it seems reasonable that the third modifiers would be a single word.

I'm going to swap to `motion_vectors` just in the interest of getting it merged, even if I think it's not an improvement.